Library/Homebrew/formulary.rb

```diff
@@ -122,14 +122,14 @@ def initialize(bottle_name)
       super name, Formulary.path(full_name)
     end
 
-    def get_formula(spec, alias_path: nil)
-      formula = super
+    def get_formula(spec, alias_path: nil) # rubocop:disable Lint/UnusedMethodArgument
+      bottle_version = Utils::Bottles.resolve_version @bottle_filename
```

Comment:
Can this fail?

Reply:
If the bottle does not contain a file matching the regex `.+/.+/INSTALL_RECEIPT.json`, then yes, it could fail. It currently throws an exception:
```
Error: undefined method `split' for nil:NilClass
/Users/sjackman/.homebrew/Library/Homebrew/utils/bottles.rb:40:in `resolve_formula_names'
```

Reply:
I've added
```diff
--- a/Library/Homebrew/utils/bottles.rb
+++ b/Library/Homebrew/utils/bottles.rb
@@ -29,9 +29,11 @@ module Utils
       end
 
       def receipt_path(bottle_file)
-        Utils.popen_read("tar", "-tzf", bottle_file).lines.map(&:chomp).find do |line|
+        path = Utils.popen_read("tar", "-tzf", bottle_file).lines.map(&:chomp).find do |line|
           line =~ %r{.+/.+/INSTALL_RECEIPT.json}
         end
+        raise "This bottle does not contain the file INSTALL_RECEIPT.json: #{bottle_file}" if path.nil?
+        path
       end
 
       def resolve_formula_names(bottle_file)
```